### DISCUSSION TITLE
FR(scm-diff-editor): have left arrow (and h) also fold sections

### DISCUSSION THREAD

**Discussion by avamsi** [arxanas/git-branchless#1051] [Ideas]
hg diff editor does this -- IMO, it's easier to press left arrow twice if you're in the middle of a section to fold it (instead of the current left arrow and then f), this would also be symmetric to right arrow (and l) unfolding sections (which is already the case).

**Comment by arxanas** [comment]
If I'm understanding correctly, if we did this, then, starting with an expanded section in some file, there would be no way using the keyboard to go to the next file without collapsing at least one section in the current file? It behaves like this for expanding/collapsing files, but there is no workflow that is impeded in this way, since you can navigate everywhere without collapsing the file.

**Reply by avamsi** [reply]
Did you mean to say go to the _previous_ file? In which case, yes, with this FR left arrow would first fold an expanded section and only then go to the previous file. If you did mean next file (I'm probably missing something), could you maybe clarify which shortcuts you have in mind?

(This is orthogonal to this FR but does provide a way of navigating around without folding sections..)
At the risk of introducing complexity, n/p could be added for next/previous section respectively and N/P for next/previous file.

**Reply by arxanas** [reply]
@avamsi If I understand your proposal correctly, then, using the keyboard only, there's no way to go from A to C without collapsing B (and children) along the way:

<img width="846" alt="Screenshot 2023-10-14 at 12 27 20" src="https://github.com/arxanas/git-branchless/assets/454057/f6d34370-65dd-424b-aaf5-023362a99631">

**Reply by avamsi** [reply]
Yep, that's right (other than the slower up arrow (and k), of course).
